dss/events/handlers/index.py

```diff
@@ -17,32 +15,52 @@
 from ...util import create_blob_key
 from ...hcablobstore import BundleMetadata, BundleFileMetadata
 from ...util.es import ElasticsearchClient
+from ...blobstore.s3 import S3BlobStore
+from ...blobstore.gs import GSBlobStore
+from ...blobstore import BlobNotFoundError
 
 DSS_BUNDLE_KEY_REGEX = r"^bundles/[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-4[0-9A-Fa-f]{3}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}\..+$"
 
 
-def process_new_indexable_object(event, logger) -> None:
+def process_new_s3_indexable_object(event, logger) -> None:
     try:
         # This function is only called for S3 creation events
         key = unquote(event['Records'][0]["s3"]["object"]["key"])
-        if is_bundle_to_index(key):
-            logger.info(f"Received S3 creation event for bundle which will be indexed: {key}")
-            s3 = boto3.resource('s3')
-            bucket_name = event['Records'][0]["s3"]["bucket"]["name"]
-            manifest = read_bundle_manifest(s3, bucket_name, key, logger)
-            bundle_id = get_bundle_id_from_key(key)
-            index_data = create_index_data(s3, bucket_name, bundle_id, manifest, logger)
-            add_index_data_to_elasticsearch(bundle_id, index_data, logger)
-            subscriptions = find_matching_subscriptions(index_data, logger)
-            process_notifications(bundle_id, subscriptions, logger)
-            logger.debug(f"Finished index processing of S3 creation event for bundle: {key}")
-        else:
-            logger.debug(f"Not indexing S3 creation event for key: {key}")
+        bucket_name = event['Records'][0]["s3"]["bucket"]["name"]
+        blobstore = S3BlobStore()
```

Comment:
I would encourage you to call `Config.get_cloud_specific_handles(..)`

Reply:
I just recently found that code. Wish i knew about it sooner. I'll make this change. 👍